.github/workflows/publish-to-ghcr.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish Docker 🐳 images 📦 to GitHub Container Registry
+
+on:
+  release:
+    types: [created]
+  workflow_dispatch: {}
+
+jobs:
+
+  build-and-publish-to-ghcr:
+    name: Build and publish Docker 🐳 images 📦 to GitHub Container Registry
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/${{ github.repository }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I'm not super familiar with github actions - what does this do?

Reply:
Good question! I've added a few inline code comments to try and explain this. I've also defined more [explicit permissions](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#assigning-permissions-to-a-specific-job) to make it hopefully a bit secure and clearer as [this token has quite far reaching permissions by default](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) 😄

Reply:
basically, this will cause the image to be pushed to github's docker registry when a release is cut. Looks good.